collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_ra.py

```diff
@@ -87,7 +89,7 @@ def disable_ra(module, array):
     """Disable Remote Assist"""
     changed = True
     if not module.check_mode:
-        if array.get_remote_assist_status()["status"] == "enabled":
+        if not array.get_remote_assist_status()["status"] in ["connected", "enabled"]:
```

Comment:
Remove `not` from `if not array.get_remote_assist_status`, since currently this will only try to disconnect if it's already disconnected. Otherwise ok.